f8a_worker/workers/CVEchecker.py

```diff
@@ -153,23 +155,24 @@ def _npm_scan(self, arguments):
     def update_depcheck_db_on_s3():
         """Update OWASP Dependency-check DB on S3."""
         s3 = StoragePool.get_connected_storage('S3VulnDB')
-        depcheck = os.path.join(configuration.OWASP_DEP_CHECK_PATH, 'bin',
-                                'dependency-check.sh')
+        depcheck = configuration.dependency_check_script_path
         with tempdir() as temp_data_dir:
             s3.retrieve_depcheck_db_if_exists(temp_data_dir)
+            old_java_opts = os.getenv('JAVA_OPTS', '')
+            os.environ['JAVA_OPTS'] = CVEcheckerTask.dependency_check_jvm_mem_limit
             # give DependencyCheck 25 minutes to download the DB
             if TimedCommand.get_command_output([depcheck, '--updateonly', '--data', temp_data_dir],
                                                timeout=1500):
                 s3.store_depcheck_db(temp_data_dir)
+            os.environ['JAVA_OPTS'] = old_java_opts
```

Comment:
What if the command throws an exception? Wouldn't be better to modify environment just for the subcommand? Like for example in https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/workers/mercator.py#L205-L209.

Reply:
Oh, this is what I was looking for but haven't found, thanks.
However the `TimedCommand.run()` works differently than `TimedCommand.get_command_output()`, so I'd rather stick with the `get_command_output()` if you don't mind :-)
It does not throw an exception, because it runs with `graceful=True`.